File: sdks/php/src/Resources/CashDrawersResource.php
<?php

namespace Puxbay\Resources;

use Puxbay\Models\CashDrawerSession;
use Puxbay\Models\PaginatedResponse;

class CashDrawersResource extends BaseResource
{
    public function list(int $page = 1): PaginatedResponse
    {
        $data = $this->get("cash-drawers/?page={$page}");
        return PaginatedResponse::fromArray($data, fn($item) => CashDrawerSession::fromArray($item));
    }

    public function getDrawer(string $drawerId): CashDrawerSession
    {
        $data = $this->get("cash-drawers/{$drawerId}/");
        return CashDrawerSession::fromArray($data);
    }

    public function open(array $drawerData): CashDrawerSession
    {
        $data = $this->post('cash-drawers/', $drawerData);
        return CashDrawerSession::fromArray($data);
    }

    public function close(string $drawerId, float $actualCash): CashDrawerSession
    {
        $data = $this->post("cash-drawers/{$drawerId}/close/", ['actual_cash' => $actualCash]);
        return CashDrawerSession::fromArray($data);
    }
}

File: sdks/dotnet/src/Resources/CashDrawersResource.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using Puxbay.SDK.Models;

namespace Puxbay.SDK.Resources
{
    public class CashDrawersResource
    {
        private readonly PuxbayClient _client;
        public CashDrawersResource(PuxbayClient client) { _client = client; }

        public async Task<PaginatedResponse<CashDrawerSession>> ListAsync(int page = 1)
        {
            return await _client.GetAsync<PaginatedResponse<CashDrawerSession>>($"cash-drawers/?page={page}");
        }

        public async Task<CashDrawerSession> GetAsync(string drawerId)
        {
            return await _client.GetAsync<CashDrawerSession>($"cash-drawers/{drawerId}/");
        }

        public async Task<CashDrawerSession> OpenAsync(Dictionary<string, object> drawerData)
        {
            return await _client.PostAsync<CashDrawerSession>("cash-drawers/", drawerData);
        }

        public async Task<CashDrawerSession> CloseAsync(string drawerId, double actualCash)
        {
            return await _client.PostAsync<CashDrawerSession>($"cash-drawers/{drawerId}/close/", new { actual_cash = actualCash });
        }
    }
}
